edgedb/_testbase.py

```diff
@@ -52,52 +54,76 @@ def flt(log_record):
 _default_cluster = None
 
 
-def _init_cluster(data_dir=None, *, cleanup_atexit=True):
-    if (not os.environ.get('EDGEDB_DEBUG_SERVER') and
-            not os.environ.get('EDGEDB_LOG_LEVEL')):
-        _env = {'EDGEDB_LOG_LEVEL': 'silent'}
-    else:
-        _env = {}
-
-    if data_dir is None:
-        cluster = edgedb_cluster.TempCluster(env=_env, testmode=True)
-        destroy = True
-    else:
-        cluster = edgedb_cluster.Cluster(data_dir=data_dir, env=_env)
-        destroy = False
-
-    if cluster.get_status() == 'not-initialized':
-        cluster.init()
-
-    cluster.start(port='dynamic')
-    cluster.set_superuser_password('test')
-
-    if cleanup_atexit:
-        atexit.register(_shutdown_cluster, cluster, destroy=destroy)
-
-    return cluster
-
-
 def _start_cluster(*, cleanup_atexit=True):
     global _default_cluster
 
-    if _default_cluster is None:
-        cluster_addr = os.environ.get('EDGEDB_TEST_CLUSTER_ADDR')
-        if cluster_addr:
-            conn_spec = json.loads(cluster_addr)
-            _default_cluster = edgedb_cluster.RunningCluster(**conn_spec)
+    if isinstance(_default_cluster, Exception):
+        # when starting a server fails
+        # don't retry starting one for every TestCase
+        # because repeating the failure can take a while
+        raise _default_cluster
+
+    if _default_cluster:
+        return _default_cluster
+
+    try:
+        tmpdir = tempfile.TemporaryDirectory()
+        status_file = os.path.join(tmpdir.name, 'server-status')
+
+        # if running on windows adjust the path for WSL
+        status_file_unix = (
+            status_file.replace('C:', '/mnt/c')
```

Comment:
It could be other drive here, so I'd go for a regexp replacement.

Reply:
changed